app/lib/ability.rb

```diff
@@ -290,6 +257,42 @@ def facility_senior_staff_abilities(user, resource, controller)
 
 
   def facility_staff_abilities(user, resource, controller)
+    can :available_for_cross_core_ordering, Product if controller.is_a?(FacilityOrdersController) || controller.is_a?(ProductsController)
+
+    if controller.is_a?(FacilityAccountsController) && user.facility_staff_of_any_facility?
```

Comment:
Use if/elsif for controller logic, consider moving cross core abilities into its own method